server/src/main/java/org/opensearch/common/io/stream/NamedWriteableRegistryResponse.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.common.io.stream;
+
+import org.opensearch.transport.TransportResponse;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Extensibility support for Named Writeable Registry: response from extensions for name writeable registry entries
+ *
+ * @opensearch.internal
+ */
+public class NamedWriteableRegistryResponse extends TransportResponse {
+
+    private final Map<String, String> registry;
+
+    public NamedWriteableRegistryResponse(Map<String, String> registry) {
+        this.registry = registry;
+    }
+
+    public NamedWriteableRegistryResponse(StreamInput in) throws IOException {
+
+        Map<String, String> registry = new HashMap<>();
+
+        // stream output for registry map begins with a variable integer that tells us the number of entries being sent across the wire
+        int registryEntryCount = in.readVInt();
+        for (int i = 0; i < registryEntryCount; i++) {
+            String name = in.readString();
+            String categoryClassName = in.readString();
+            registry.put(name, categoryClassName);
+        }
+
+        this.registry = registry;
+    }
+
+    public Map<String, String> getRegistry() {
+        return this.registry;
```

Comment:
This returns your "final" map and the caller could change it.  Either make a copy or wrap in `Collections.unmodifiableMap()`.